llvm/lib/Target/SPIRV/SPIRVBlockLabeler.cpp

```diff
@@ -116,25 +113,15 @@ bool SPIRVBlockLabeler::runOnMachineFunction(MachineFunction &MF) {
       if (MI.getOpcode() == SPIRV::OpBranch) {
         Branches.push_back(&MI);
       } else if (MI.getOpcode() == SPIRV::OpBranchConditional) {
-        // If previous optimizations generated conditionals with implicit
-        // fallthrough, add the missing explicit "else" to it valid SPIR-V
-        if (MI.getNumOperands() < 3) {
-          MI.addOperand(MachineOperand::CreateMBB(MBB.getNextNode()));
-        }
+        assert(MI.getNumOperands() == 3 &&
```

Comment:
The spec says it's possible we may have >3 operands for OpBranchCond, I agree we're currently processing only 3 of them though

Reply:
Yes, we don't create OpBranchConditional with 5 operands (3 normal+2 branch weights). We can add TODO comment about this spec case here.